src/SALib/analyze/delta.py

```diff
@@ -130,13 +144,16 @@ def calc_delta(Y, Ygrid, X, m):
     return d_hat
 
 
-def bias_reduced_delta(Y, Ygrid, X, m, num_resamples, conf_level):
+def bias_reduced_delta(Y, Ygrid, X, m, num_resamples, conf_level, Nr):
     """Plischke et al. 2013 bias reduction technique (eqn 30)"""
     d = np.empty(num_resamples)
-    d_hat = calc_delta(Y, Ygrid, X, m)
 
     N = len(Y)
-    r = np.random.randint(N, size=(num_resamples, N))
+    if Nr is None: 
+        Nr = N
+    ind = np.random.randint(N, size=Nr)
+    d_hat = calc_delta(Y[ind], Ygrid, X[ind], m)
+    r = np.random.randint(N, size=(num_resamples, Nr))

```

Comment:
I suppose we don't need to handle the None case as this function is more or less private.
```suggestion
    ind = np.random.randint(N, size= y_resamples)
    d_hat = calc_delta(Y[ind], Ygrid, X[ind], m)
    r = np.random.randint(N, size=(num_resamples, y_resamples))

```